compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -408,6 +421,12 @@ private void visitType(Location loc, BType expType, BType actualType, SymbolEnv
                     findTypeParamInInvokableType(loc, (BInvokableType) expType, (BInvokableType) actualType, env,
                             resolvedTypes, result);
                 }
+                if (actualType.tag == TypeTags.SEMANTIC_ERROR) {
+                    if (expr != null && expr.getKind() == NodeKind.LAMBDA) {
```

Comment:
Can we merge these two` if`'s to one?

Reply:
Yes, we can, but I think, having it like this will be more readable. It is something like,
```ballerina
if (actualType.tag == TypeTags.SEMANTIC_ERROR) { // type-tag check
        // do something
}
```
WDYT?

Reply:
They both read pretty much the same though?